packages/worker/src/workflow/reusable-vm.ts

```diff
@@ -0,0 +1,250 @@
+import assert from 'assert';
+import { AsyncLocalStorage } from 'async_hooks';
+import vm from 'vm';
+import { gte } from 'semver';
+import * as internals from '@temporalio/workflow/lib/worker-interface';
+import { IllegalStateError } from '@temporalio/common';
+import { Workflow, WorkflowCreateOptions, WorkflowCreator } from './interface';
+import { WorkflowBundleWithSourceMapAndFilename } from './workflow-worker-thread/input';
+import { BaseVMWorkflow, globalHandlers, setUnhandledRejectionHandler } from './vm-shared';
+
+// Thanks MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
+function deepFreeze(object: any) {
+  // Retrieve the property names defined on object
+  const propNames = Object.getOwnPropertyNames(object);
+
+  // Freeze properties before freezing self
+  for (const name of propNames) {
+    const value = object[name];
+
+    if (value && typeof value === 'object') {
+      try {
+        deepFreeze(value);
+      } catch (err) {
+        // This is okay, there are some typed arrays that cannot be frozen (encodingKeys)
+      }
+    } else if (typeof value === 'function') {
+      Object.freeze(value);
+    }
+  }
+
+  return Object.freeze(object);
+}
+
+/**
+ * A WorkflowCreator that creates VMWorkflows in the current isolate
+ */
+export class ReusableVMWorkflowCreator implements WorkflowCreator {
+  /**
+   * TODO(bergundy): Get rid of this static state somehow
```

Comment:
I dunno how you could ever get rid of `unhandledRejectionHandlerHasBeenSet` since it's for a process-wide handler.

The map, maybe.